odoo/fields.py

```diff
@@ -577,12 +581,18 @@ def _compute_related(self, records):
         """ Compute the related field ``self`` on ``records``. """
         # when related_sudo, bypass access rights checks when reading values
         others = records.sudo() if self.related_sudo else records
-        for record, other in zip(records, others):
-            if not record.id and record.env != other.env:
-                # draft records: copy record's cache to other's cache first
-                copy_cache(record, other.env)
-            other, field = self.traverse_related(other)
-            record[self.name] = other[field.name]
+        # copy new records' cache into others' cache
+        if records.env != others.env:
+            copy_cache(records - records.filtered('id'), others.env)
+        # traverse fields one by one for all records, in order to take advantage
+        # of prefetching at each level. This is important specially for the last
+        # field, in case it is a computed field!
```

Comment:
I don't think the comment is very clear as to what happens unless you already know. I had an idea of the purpose and I still had a hard time understanding it.

Also should explain *why* it's a big deal if the final field of the relation chain is computed, compared to resolving the entire "related chain" for each record.

Reply:
Thanks for pointing this out. I am going to clarify this!